pkg/util/docker/docker_test.go

```diff
@@ -92,86 +92,337 @@ func TestResolveImageName(t *testing.T) {
 }
 
 func TestParseContainerNetworkAddresses(t *testing.T) {
-	ports := []types.Port{
+	for i, tc := range []struct {
+		name        string
+		ports       []types.Port
+		netSettings *types.SummaryNetworkSettings
+		expected    []containers.NetworkAddress
+	}{
 		{
-			IP:          "0.0.0.0",
-			PrivatePort: 80,
-			PublicPort:  8080,
-			Type:        "tcp",
+			name:        "empty input",
+			ports:       []types.Port{},
+			netSettings: nil,
+			expected:    []containers.NetworkAddress{},
 		},
 		{
-			PrivatePort: 80,
-			Type:        "tcp",
+			name: "exposed port",
+			ports: []types.Port{
+				{
+					IP:          "0.0.0.0",
+					PrivatePort: 80,
+					PublicPort:  8080,
+					Type:        "tcp",
+				},
+			},
+			netSettings: &types.SummaryNetworkSettings{
+				Networks: map[string]*network.EndpointSettings{
+					"bridge": {
+						IPAMConfig:          nil,
+						Links:               nil,
+						Aliases:             nil,
+						NetworkID:           "NetworkID",
+						EndpointID:          "EndpointID",
+						Gateway:             "172.17.0.1",
+						IPAddress:           "172.17.0.2",
+						IPPrefixLen:         16,
+						IPv6Gateway:         "",
+						GlobalIPv6Address:   "",
+						GlobalIPv6PrefixLen: 0,
+						MacAddress:          "MacAddress",
+					},
+				},
+			},
+			expected: []containers.NetworkAddress{
+				{
+					IP:       net.ParseIP("0.0.0.0"),
```

Comment:
I thoufght what we discussed in this case was getting the host IP an not 0.0.0.0

Reply:
But if it's not mentioned, docker default the value it to `0.0.0.0`. Do you suggest not reporting it if it's `0.0.0.0` ?

Reply:
@leeavital I think we can mark this as resolved based on our discussion